### PR TITLE
MRG: avoid warning in SURE when passing grid & allow to return empty stc

### PR DESCRIPTION
**Pull request by agramfort**
cc @PABannier 

let me know what you think

### CODE REVIEW

File: mne/inverse_sparse/mxne_inverse.py [hunk 1]
```diff
@@ -441,8 +442,9 @@ def mixed_norm(evoked, forward, noise_cov, alpha='sure', loose='auto',
 
     # Alpha selected automatically by SURE minimization
     if alpha == "sure":
-        alpha_grid = (np.geomspace(100, 10, num=15)
-                      if sure_alpha_grid == "auto" else sure_alpha_grid)
+        alpha_grid = sure_alpha_grid
+        if isinstance(sure_alpha_grid, str) and sure_alpha_grid == "auto":
+            alpha_grid = np.geomspace(100, 10, num=15)
```
Comment (PABannier) [inline mne/inverse_sparse/mxne_inverse.py:447]: I don't know if it still makes sense to use np.geomspace, compared to np.arange. np.geomspace results in a sample with a lot of values near 10, and fewer around 100. I'm not sure it is what we want. WDYT?

**Comment by PABannier** [comment]
For the empty STC + warning part, LGTM!

File: mne/inverse_sparse/mxne_inverse.py (hunk 1 above)
Comment (agramfort) [reply mne/inverse_sparse/mxne_inverse.py:447]: you mean np.linspace? can you test if np.linspace works better on some datasets @PABannier ?

Comment (PABannier) [reply mne/inverse_sparse/mxne_inverse.py:447]: yes np.linspace. I have already tried it. I did not notice any change in performance. in some ways, np.linspace does encourage sparse solutions and might be faster.

**Pull request merged**